connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -628,8 +625,13 @@ object SparkSession extends Logging {
   /**
    * Create a new [[SparkSession]] based on the connect client [[Configuration]].
    */
-  private[sql] def create(configuration: Configuration): SparkSession = {
-    new SparkSession(new SparkConnectClient(configuration), cleaner, planIdGenerator)
+  private[sql] def create(
+      configuration: Configuration,
+      channelBuilder: Option[ChannelBuilder] = None): SparkSession = {
```

Comment:
We can, I am open to discuss this.

The reason I didn't do it this way, is that it would make a bidirectional dependency between Configuration and ChannelBuilder.

Reply:
What is currently part of configuration and why should the ChannelBuilder be part of it? Is there a meaningful delta between them?

Reply:
https://github.com/apache/spark/blob/7fcabef28743099363c5cf21b90e987e6be90b12/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala#L466

ChannelBuilder could be part of it because it's also part of SparkConnectClient's configuration